src/foms.py

```diff
@@ -0,0 +1,186 @@
+            subset    = int(matches.group(2))
+            foms = get_foms(commandStart + str(image_file_name))
+            crcs, snrs = foms
+            write(outfile,cli_args, f'{iteration:02}-{subset:02} ', end='')
+            crcs, variabilities = tuple(zip(*((crc, var) for (r, (crc, var)) in crcs.items())))
+            write(outfile,cli_args, ''.join(f'{c:6.1f} ' for c in crcs)         , end='         ')
+            write(outfile,cli_args, ''.join(f'{v:6.1f} ' for v in variabilities), end='         ')
+            write(outfile,cli_args, ''.join(f'{r:6.1f} ' for r in snrs.values())) # look broken in the Rust implementation of foms
+    
+def plot_from_fom(dir, sphere_diameters, cli_args):
+    count = 0
+    isData = False
+    data = dict()
+    for d in sphere_diameters:
+        data[d] = [] #initialize lists for each diameter key
+    with open(dir, encoding='utf-8') as f:
+        Lines = f.readlines()
+        for line in Lines:
+            if count == 4:
+                isData = True
+            if not isData:
+                count += 1
+                continue
+            #remove first element from a line and convert the remaining elements to floats
+            parsedLine = list(map(float, line.split()[1:]))
+            for i in range(len(sphere_diameters)):
+                data[sphere_diameters[i]].append((float(count-4), parsedLine[i],
+                                                  parsedLine[i+len(sphere_diameters)]))
+            count += 1
+    for d in sphere_diameters:
+        x = [data[d][i][0] for i in range(len(data[d]))]
+        y = [data[d][i][1] for i in range(len(data[d]))]
+        e = [data[d][i][2] for i in range(len(data[d]))]
+        plt.figure()
+        plt.plot(x,y,linewidth=2.0)
+        plt.errorbar(x,y,yerr=e,label=f'TOF=t c',capsize=3)
+        plt.legend()
+        plt.ylim(top=120)
+        plt.title(f'CRC vs iteration for {d}mm sphere')
+        plt.savefig(f'{cli_args["<DIR>"]}/crcs-{d}mm.png')
+
+    if cli_args['--show-plot']:
+        plt.show()
+
+def main():
+    cli_args = docopt(__doc__)
+    directory = cli_args['<DIR>']
+    print(f'Calculating FOMs for images in {directory}')
+    filename = f'{directory}/foms'
+    images = sorted(Path(directory).glob('*.raw'))
+
+    if bool(cli_args['<jaszczak>']):
+        commandStart = 'target/release/foms -- jaszczak '
+        write_command(filename, images, commandStart, jaszczak_sphere_diameters,
+                      cli_args)
+        plot_from_fom(filename, jaszczak_sphere_diameters, cli_args)
+
+    elif bool(cli_args['<nema7>']):
+        commandStart = 'target/release/foms -- nema7 '
+        write_command(filename, images, commandStart, nema7_sphere_diameters,
+                      cli_args)
+        plot_from_fom(filename, nema7_sphere_diameters, cli_args)
+
+    else:
+        print("You must specify a known phantom type: (jaszczak/nema7)")
+
+if __name__=='__main__':
+   main()
```

Comment:
I'm no fan of these pointless guards which add no useful functionality of the code.

I'm aware that > 90% of Python programmers stick these in by default, and some (many?, most?) IDEs do too, but that's not a good enough reason for me.

Do you know what this does? What it's purpose is?

These are useful when you want the same source file to be usable both as a module (something you `import`) and as program (something you 'execute'). This construct allows you to distinguish between the two situations from within the code, and do things slightly differently. In brief

+ The program should probably do some work, produce some output, etc.
+ The module should most probably *not* produce any output or do any work, besides defining the utilities that the `import`er wants to use.

More official details here: https://docs.python.org/3/library/__main__.html

This particular source file is only intended to be used as a program, and defines nothing that might be useful to import elsewhere, so let's not pretend that it's otherwise by including this guard.

I'm also no fan of pointless `main` functions. Python programs are executed from top to bottom, so you don't need a `main` to tell the system what the entry point is (unlike C, Rust, etc.).

I'm less negative about pointless `main`s than I am about pointless `__main__` guards, as there might be *some* value in grouping or hiding stuff in functions. You don't pollute the global scope with `main`'s local variables, for one.

In short, you can keep `main()`, if you like, but please remove the guard.